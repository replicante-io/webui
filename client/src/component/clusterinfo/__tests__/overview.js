'use strict';
//@flow

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Overview from '../overview';


const mockStore = configureStore([]);
const CLUSTER = {
  agents_down: 0,
  cluster_display_name: 'test',
  cluster_id: 'test',
  kinds: ['Kafka'],
  nodes: 3,
  nodes_down: 1,
  shards_count: 2,
  shards_primaries: 1,
};
const MATCH = {
  params: {name: 'test'}
};


describe('ClusterInfo', () => {
  describe('overview', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusterinfo: {
          meta: {}
        }
      });
    });

    test('renders with data', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <Overview cluster={CLUSTER} match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
