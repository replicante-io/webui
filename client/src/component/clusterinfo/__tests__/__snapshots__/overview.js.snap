// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterInfo overview renders with data 1`] = `
<div>
  <div>
    Cluster type: 
    <span
      className="db-icon kafka"
      title="Kafka"
    />
  </div>
  <div>
    Total number of shards in the cluster: 
    2
  </div>
  <div>
    Total number of nodes in the cluster: 
    3
    <ul>
      <li>
        Of which have an agent down: 
        0
      </li>
      <li>
        Of which are down: 
        1
      </li>
    </ul>
  </div>
</div>
`;
